tt_metal/hw/inc/dataflow_api.h

```diff
@@ -582,8 +581,11 @@ void noc_async_read_one_packet_with_state(std::uint32_t src_noc_addr, std::uint3
 
     DEBUG_STATUS("NARW");
 
-    // TODO: need a way sanitize size + addr w/o directly providing x/y here (grab x/y form state?)
-    // DEBUG_SANITIZE_READ_NOC_TRANSACTION(src_noc_addr, dst_local_l1_addr, size);
+    // In order to sanitize, need to grab full noc addr + xfer size from state.
+    std::uint64_t src_noc_addr_full = NOC_CMD_BUF_READ_REG(noc_index, NCRISC_RD_CMD_BUF, NOC_TARG_ADDR_MID);
```

Comment:
These need to be wrapped in an ifdef - they are volatile register reads which the compiler can't optimize away

Reply:
Added an ifdef here, new CI running: https://github.com/tenstorrent/tt-metal/actions/runs/9084172911

Reply:
sorry, I should have been more clear, I'm going to nitpick: to keep the mainline code clean/simple, let's create a DEBUG_SANITIZE_NOC_READ_XYZSTATESOMETHING() that handles this and goes away w/ watcher disabled.  Generally I like to push ifdefs down into inline fns or macros so it is easy to see the flow at the top level

Reply:
Makes sense, updated this. New CI: https://github.com/tenstorrent/tt-metal/actions/runs/9100552147